### PR TITLE
Feat/pii subprocess venv

### PR DESCRIPTION
**Pull request by wilke0818**
## Summary
  
  Make PII detection actually run on Python 3.14 hosts (it was silently no-op'ing because spaCy / Presidio top out at cp313 wheels), and pass through richer per-detection
  signal to consumers instead of just a single boolean.

  Three logical changes stacked, each in its own commit so they're individually reviewable and revertible:

  1. **`91ec2327`** — Move PII detection into a new `pii-detection` subprocess venv (Python 3.13) running Presidio + GLiNER (`nvidia/gliner-pii`). Subprocess-only, no
  in-process fallback. Per-detector opt-in via a `detectors=[...]` argument symmetrical to ASR-model selection elsewhere.
  2. **`4129b49d`** — Add `detection_confidence` to `PiiPassReport`: a continuous `[0, 1]` aggregated from per-detector scores × cross-detector agreement × cross-ASR-model
  agreement. Replaces the prior `1.0 if contains_pii else 0.0` boolean in `global_summary`. `None` propagates when no detector actually ran (subprocess crash, both detectors
  failed to load, or caller passed `detectors=[]`) — distinct from `0.0` ("ran, found nothing").
  3. **`a2f384ce` + `628973cc`** — GLiNER default labels set to the HIPAA Safe Harbor 18 identifiers verbatim, matching b2aiprep PR #256 so the two projects' PII outputs share
  a vocabulary.

  ## Why a subprocess at all

  The host venv may be on a Python version newer than spaCy ships wheels for (3.14 today). Pre-this-PR the host-side detector chain silently fell back to
  `failures["no_pii_detector"]` + `contains_pii: False` on every clip — a privacy-meaningful false negative on b2ai voice data. Moving Presidio + GLiNER into their own
  Python-3.13 subprocess venv (built and cached via `ensure_venv`) keeps the host on whatever recent Python it likes while making the PII path actually run.
  
  ## What runs inside the venv

  Two detectors run on every transcript and their spans are merged before cross-ASR-model corroboration in `pii.py`:

  - **Microsoft Presidio Analyzer** — regex + spaCy-NER recognizers for emails, phone numbers, SSNs, credit cards, IPs, dates, locations.
  - **GLiNER PII** (`nvidia/gliner-pii`) — a 570M-param transformer fine-tuned on ~100k synthetic PII/PHI records. Catches HIPAA PHI categories Presidio doesn't natively
  recognize (`medical_record_number`, `health_plan_number`, `account_number`, `fax_number`, `url`, `biometric_identifier`, `unique_identifier`, ...).

  GLiNER's lowercase labels are normalized to Presidio's uppercase scheme inside the worker so the existing `(category, text.lower())` dedupe key in `pii.py` treats both
  detectors' hits on the same entity as the same finding.

  ## Per-detector opt-in

  ```python
  # Default — both detectors
  detect_pii_in_pass(pass_label="raw_16k", asr_resolved=...)

  # Presidio only — skip GLiNER's ~5 s model load
  detect_pii_in_pass(..., detectors=["presidio"])

  # GLiNER only — skip Presidio
  detect_pii_in_pass(..., detectors=["gliner"])

  # Explicit disable — no subprocess spawn, no venv build attempt;
  # report carries failures["pii_disabled"] so an auditor can tell
  # "didn't run on purpose" apart from "ran and found nothing"
  # and "subprocess crashed"
  detect_pii_in_pass(..., detectors=[])
```

  gliner_model, gliner_labels, and gliner_threshold are also exposed on detect_pii_in_pass so callers can swap in a different GLiNER variant (e.g. urchade/gliner_multi-v2.1 for
   multilingual coverage) or restrict the label set without touching the worker.

  detection_confidence vs contains_pii

  The PII axis previously surfaced only contains_pii ? 1.0 : 0.0 to global_summary and threw away every per-span detector score the worker produced. Worse, when the subprocess
  crashed or no detector loaded, the report fell to 0.0 — indistinguishable from "we checked, found nothing."

  detection_confidence is now a continuous [0, 1] combining three signals per unique (category, normalized_text) finding:

  - per-span detector score — Presidio analyzer score or GLiNER prediction probability, max() within the finding's group.
  - cross-detector agreement — Presidio AND GLiNER both flagged the same finding (factor 1.0) vs only one detector (0.5). The strongest "real entity vs hallucination?" signal
  at this layer.
  - cross-ASR-model agreement — fraction of available ASR transcripts that contained the finding. Spans only one ASR transcribed and no sibling ASR confirms are the
  prototypical hallucination case.

  Then max() across findings — matches how the transcript and single-speaker axes already combine their internal sub-signals.

  Deliberately NO category-severity weighting. In pediatric / clinical voice data the categories nominally most "severe" (US_SSN, CREDIT_CARD, MEDICAL_RECORD_NUMBER) have
  near-zero true-positive rate — children don't speak SSNs aloud — and the synthetic-PII-trained detectors gladly flag any 9-digit ASR hallucination as SSN. Weighting them up
  would inflate exactly the findings a reviewer should de-prioritize. Heuristic enhancements deferred as later goals.
  
  None propagation

  detection_confidence is None when no detector ran:

  - subprocess raised at dispatch (CalledProcessError / RuntimeError)
  - both detectors failed to load inside the subprocess venv
  - caller passed detectors=[] (explicit-disable short-circuit)

  0.0 is reserved for "detectors ran, found nothing" — the confident-negative case. global_summary.compute_pass_global_summary propagates the same distinction into
  no_pii_uncertainty so the combined_uncertainty consumer can tell "didn't actually check" apart from "checked clean." No silent demotion to 0.

  Why the HIPAA-18 labels (and why exactly that list)

  GLiNER picks were tuned via an out-of-band diagnostic (temp/diag_gliner_email.py) on john.doe@example.com. Two non-obvious findings drove the final label set:

  1. nvidia/gliner-pii exhibits competing-claim interference. When two labels can plausibly cover the same span, the model commits the span to one and silently drops it from
  consideration under the other — even when the other would have been the right answer. Passing [person, first_name, last_name, email, email_address, ...] lost the email
  entirely; the model classified John and Doe as first_name/last_name at score 1.0 and emitted no span containing @. A flat label set with one name and one email caught both at
   score 1.0. Keep label sets flat.

  2. Even flat label sets have a length / projection-space effect we don't fully understand. Tacking the two clinical-voice extensions health_condition and medication onto the
  end of the HIPAA-18 list (semantically non-overlapping with anything) was enough to push the email's score below the 0.5 threshold and lose it. The HIPAA-18 verbatim catches
  it cleanly. Don't extend past the HIPAA-18 unless you've verified your specific additions don't trip the interference on your corpus.
  
  Both findings are captured as DO NOT warnings above _DEFAULT_GLINER_LABELS so a future contributor doesn't walk back into either trap.

  Architectural changes
  
  - New module src/senselab/audio/workflows/audio_analysis/pii_subprocess.py — venv constants (_PII_VENV = "pii-detection", Python 3.13, presidio-analyzer + spacy +
  en_core_web_lg wheel via direct URL + gliner + transformers + torch), GLiNER → Presidio category map, worker script, and the detect_pii_via_subprocess dispatch function.
  torch named explicitly in _PII_REQUIREMENTS so the auto-detect in ensure_venv routes it through the matched CUDA index. torchaudio intentionally omitted (no audio decoding).
  - pii.py simplified — removed _load_presidio_analyzer, _load_spacy_nlp, _scan_with_presidio, _scan_with_spacy, and the in-process tier-fallback chain. Dispatch now goes to
  the subprocess; everything else (transcript concatenation, PiiSpan materialization, cross-ASR-model corroboration, PiiPassReport construction) stays in-process.
  - global_summary.compute_pass_global_summary reads pii_report.detection_confidence instead of the boolean for the no_pii_uncertainty field. Adds a pii_report.detector_used is
   None branch so the "didn't actually run" signal surfaces as None, not silently 0.0. New detection_confidence field also emitted in the per-pass no_pii block for downstream
  tools.

  Performance notes

  - First subprocess invocation per host: ~60–120 s for the venv install (torch + nvidia-cuda-runtime + transformers are the heavy parts; everything else is small) plus ~30 s
  for the first GLiNER model download from HuggingFace.
  - Subsequent invocations on the same host: ~10 s cold-start per call (Presidio + spaCy + GLiNER model load).
  - For a typical analyze_audio.py run (≈138 detect_pii_in_pass calls per subject), that's ~15–20 min of model-load overhead added per subject. Batching all of a subject's
  transcripts into a single subprocess invocation is a worthwhile follow-up but isn't required for correctness.

  Tests
  
  32 tests added across two new files:

  - src/tests/audio/workflows/audio_analysis/pii_subprocess_test.py (10 tests) — dispatch-surface coverage: detector selection, JSON-request construction, response parsing,
  empty-detector short-circuit, unknown-detector ValueError, dedupe-with-order, default-arg pass-through, caller overrides.
  - src/tests/audio/workflows/audio_analysis/pii_test.py (12 tests) — _compute_detection_confidence unit cases (empty → 0.0, single detector / single ASR → 0.5 × score, both
  detectors → 1.0 factor, cross-ASR scaling by fraction, max across independent findings, whitespace-only text filtered, missing per-span score collapses to 0) and
  detect_pii_in_pass integration with a mocked detect_pii_via_subprocess (happy path, no-detectors → None, subprocess raise → None, ran-no-spans → 0.0, detectors=[]
  short-circuit → None plus failures["pii_disabled"]).
  
  All 32 pass. Pre-commit hooks (ruff, mypy, codespell) clean.

  The real Presidio + GLiNER worker inside the subprocess venv is exercised by the out-of-band scripts temp/test_pii_subprocess.py and temp/diag_gliner_email.py — including
  them in the unit test suite would require building the actual ~5 GB venv on every test run.

  Test plan

  - uv run pytest src/tests/audio/workflows/audio_analysis/pii_subprocess_test.py src/tests/audio/workflows/audio_analysis/pii_test.py --noconftest — 22 / 22 pass.
  - temp/test_pii_subprocess.py smoke test on ORCD: builds the venv, both Presidio and GLiNER load, both detectors flag known PII strings, detection_confidence aggregates
  correctly, all four detectors=[...] selection paths behave as designed.
  - temp/diag_gliner_email.py diagnostic confirms nvidia/gliner-pii catches john.doe@example.com at score 1.0 with the flat HIPAA-18 set and does not catch it with overlapping
  name-related labels (the empirical justification for the label-set choice).
  - Run full analyze_audio.py on a real b2ai subject and confirm per-clip pii.json files now show non-None detector_used + non-None detection_confidence.

  Out-of-band reviewer notes

  - Cold-start cost per call is real and observable. If it turns out to dominate runtime on production-size subjects, the natural follow-up is to batch all transcripts for one
  subject into a single subprocess invocation at the analyze_audio.py level rather than per-pass-per-clip. The dispatch function already accepts an arbitrary number of
  (asr_model, full_text) entries, so the change would be at the call site, not the API.
  - GLiNER label set is intentionally narrow. Callers who need a wider label set can pass gliner_labels=[...] to detect_pii_in_pass; the contributor warnings above
  _DEFAULT_GLINER_LABELS capture the two empirical traps (overlapping labels, extending past HIPAA-18) so the caller can verify their additions don't trip them.
  - No category-severity weighting is a deliberate choice for the pediatric / clinical voice domain, where the most-"severe" Presidio categories (US_SSN, CREDIT_CARD) have the
  highest false-positive rate. Reintroducing severity weighting for a different downstream domain should be opt-in, not the default.

# Version

Published prerelease version: `1.3.1-alpha.32`

<details>
  <summary>Changelog</summary>

  #### 🐛 Bug Fix
  
  - Feat/pii subprocess venv [#519](https://github.com/sensein/senselab/pull/519) ([@wilke0818](https://github.com/wilke0818))
  - subprocess_venv: route torch/torchaudio via CUDA-aware PyTorch index [#516](https://github.com/sensein/senselab/pull/516) ([@satra](https://github.com/satra) [@wilke0818](https://github.com/wilke0818))
  - subprocess_venv: two-stage install to fix torch/torchaudio CUDA-tag s… [#518](https://github.com/sensein/senselab/pull/518) ([@wilke0818](https://github.com/wilke0818))
  - Fix/ser wav2vec2 regression head [#511](https://github.com/sensein/senselab/pull/511) ([@satra](https://github.com/satra) [@wilke0818](https://github.com/wilke0818) [@claude](https://github.com/claude))
  - comparison & uncertainty stage for analyze_audio.py [#512](https://github.com/sensein/senselab/pull/512) ([@satra](https://github.com/satra))
  - audio analysis script + multilingual MMS aligner + 4 new ASR backends [#510](https://github.com/sensein/senselab/pull/510) ([@satra](https://github.com/satra))
  
  #### Authors: 3
  
  - Claude ([@claude](https://github.com/claude))
  - Jordan Wilke ([@wilke0818](https://github.com/wilke0818))
  - Satrajit Ghosh ([@satra](https://github.com/satra))
</details>

### CODE REVIEW

File: src/senselab/audio/workflows/audio_analysis/pii_subprocess.py [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+# transitive resolve in Stage 2 would skip the CUDA-aware routing and
+# could land on a CPU-only wheel even on GPU hosts. ``torchaudio`` is
+# NOT in this list — neither Presidio nor GLiNER decode audio.
+_PII_REQUIREMENTS = [
+    "presidio-analyzer>=2.2",
+    "spacy>=3.7,<3.9",
+    _EN_CORE_WEB_LG_WHEEL,
+    "gliner>=0.2",
+    "transformers>=4.40",
+    "torch>=2.8,<2.9",
```
[the remark this answers is not in the document]
Comment (wilke0818) [reply src/senselab/audio/workflows/audio_analysis/pii_subprocess.py:83]: bad bot, out of date information
